corehq/messaging/scheduling/scheduling_partitioned/dbaccessors.py

```diff
@@ -157,6 +183,24 @@ def get_case_timed_schedule_instances_for_schedule(case_id, schedule):
     return get_case_timed_schedule_instances_for_schedule_id(case_id, schedule.schedule_id)
 
 
+def get_case_schedule_instance(cls, case_id, schedule_instance_id):
+    from corehq.messaging.scheduling.scheduling_partitioned.models import (
+        CaseAlertScheduleInstance,
+        CaseTimedScheduleInstance,
+    )
+
+    if cls not in (CaseAlertScheduleInstance, CaseTimedScheduleInstance):
+        raise ValueError("Expected CaseAlertScheduleInstance or CaseTimedScheduleInstance")
```

Comment:
my thought process here was that we're checking that `cls` has one of these values, not that an object has a particular type. i don't feel strongly though. i updated it to use TypeError here https://github.com/dimagi/commcare-hq/pull/16427/commits/8cc100e8a101bf2c923923bcedd7686854090f20